en_US/rule/rule-engine.md

```diff
@@ -79,12 +79,12 @@ SELECT * FROM "$events/client_connected"
 ```
 
 ::: tip
-The old version of SQL syntax conversion function is provided in Dashboard to complete SQL upgrade and migration. 
+A conversion function that converts from the old version of the SQL syntax to the new version is provided in the Dashboard to ease SQL upgrade and migration. 
 :::
 
 ## Rule engine composition
 
-EMQX Broker's rule engine can be used to flexibly process messages and events. By using the rule engine, it can easily achieve such function as converting the message into a specified format, and then stored in a database table, or sent to the message queue.
+EMQX Broker's rule engine can be used to flexibly process messages and events. By using the rule engine, it can easily achieve such functions as converting the message into a specified format, and then stored in a database table, or sent to the message queue.
```

Comment:
```suggestion
EMQX Broker's rule engine can be used to flexibly process messages and events. By using the rule engine, it can easily achieve such functionalities as converting the message into a specified format, and then stored in a database table, or sent to the message queue.
```
?